onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -8764,6 +8764,11 @@ def aten_type_as(self: TTensor, other: TTensor2) -> TTensor2:
 def aten_unbind(self: TTensor, dim: int = 0) -> Sequence[TTensor]:
     """unbind.int(Tensor(a -> *) self, int dim=0) -> Tensor(a)[]"""
 
+    if isinstance(self.shape[dim], int):
+        # We can create a definitive split op if the input shape is static
+        outputs = op.Split(self, axis=dim, num_outputs=self.shape[dim])
+        return [op.Squeeze(out, [dim]) for out in outputs]
```

Comment:
I feel like it makes more sense to rewrite it? Although this PR probably works, it's adding another dimension on torchlib (covering both static and dynamic cases). Maybe let torchlib be as dynamic as possible, and we can optimize it after.

Reply:
How easy is it to create the optimization rules? I am fine either way

Reply:
Looks like `SplitToSequence(self, axis=dim, keepdims=False)` should be generically rewritable to the subgraph if the split axis is known. This can potentially cover more cases with other ops when encountered in the future

Reply:
But my understanding is that the original implementation is not incorrect. It's only the op is not preferred because of the backend implementation, which fits the category of rewritten rules. We surely can say it's more convenient to address this way (this PR), but I prefer an established/explicit rule to say when/what we should add support in torchlib, and under what condition we add rewrite rules. Otherwise, it's just scattered round. And if we want it to be done in this way, do we consider upstream some other optimizations downstream that are optimized away because of static shapes as well?

Reply:
I kinda think we should do both. (1) we can measure the complexity of the torchlib implementation. I feel that the complexity of the current implementation is not high for the immediate benefits it brings. If the graph can be significantly simplified because we know some shapes are static, I think we should pursue that in torchlib. When looking at micro-optimizations like this I agree that we should be more careful and decide on a case by case basis. (2) we should still have a rule that will simplify this so that our tooling can handle `SplitToSequence` generally.